pkg/controller/controller_instance.go

```diff
@@ -1389,10 +1390,31 @@ func (c *controller) prepareRequestHelper(instance *v1beta1.ServiceInstance, ser
 	rh.ns = ns
 
 	if setInProgressProperties {
-		parameters, parametersChecksum, rawParametersWithRedaction, err := prepareInProgressPropertyParameters(
+		// If the current operation is already set, we need to ignore the spec and read parameters from the status
+		var specParameters *runtime.RawExtension
+		var planExternalName string
+		var planExternalID string
+		var userInfo *v1beta1.UserInfo
+
+		if instance.Status.CurrentOperation == "" {
+			specParameters = instance.Spec.Parameters
+			planExternalName = servicePlan.Spec.ExternalName
+			planExternalID = servicePlan.Spec.ExternalID
+			userInfo = instance.Spec.UserInfo
+		} else {
+			if instance.Status.InProgressProperties == nil {
+				return nil, fmt.Errorf("InProgressProperties must not be nil if the CurrentOperation is set")
+			}
+			specParameters = instance.Status.InProgressProperties.InlineParameters
```

Comment:
This field will be empty if the operation has started prior to upgrading to the version of Service Catalog with `InlineParameters` field.

Reply:
Good question.